src/polyhedralGravity/model/KDTree/KdDefinitions.h

```diff
@@ -169,10 +308,114 @@ namespace polyhedralGravity {
          * @return true if this should precede the other argument.
          */
         bool operator<(const PlaneEvent &other) const {
-            if (this->plane.axisCoordinate == other.plane.axisCoordinate) {
+            if (this->plane.axisCoordinate != other.plane.axisCoordinate) {
+                return this->plane.axisCoordinate < other.plane.axisCoordinate;
+            }
+            if (this->plane.orientation == other.plane.orientation) {
                 return this->type < other.type;
             }
-            return this->plane.axisCoordinate < other.plane.axisCoordinate;
+            return static_cast<unsigned>(this->plane.orientation) < static_cast<unsigned>(other.plane.orientation);
+        }
+
+        /**
+         *Equality operator used for testing purposes
+         */
+        bool operator==(const PlaneEvent &other) const {
+            return type == other.type && plane == other.plane && faceIndex == other.faceIndex;
         }
     };
+
+    /**
+     * A list of PlaneEvents.
+    */
+    using PlaneEventList = std::vector<PlaneEvent>;
+
+    /**
+    * An array of PlaneEventLists.
+    */
+    template<size_t Number>
+    using PlaneEventLists = std::array<std::unique_ptr<PlaneEventList>, Number>;
+
+    /**
+     * Extracts the indices of the triangle faces that are referenced by the given PlaneEvents.
+     * @param events The PlaneEvents containing information about the faces.
+     * @return A list of face indices.
+     */
+    static TriangleIndexList convertEventsToFaces(const std::variant<TriangleIndexList, PlaneEventList> &events) {
+        if (std::holds_alternative<TriangleIndexList>(events)) {
+            return std::get<TriangleIndexList>(events);
+        }
+        const auto &eventList{std::get<PlaneEventList>(events)};
+        TriangleIndexList triangles{};
+        triangles.reserve(eventList.size());
+        //used to avoid duplication
+        std::unordered_set<unsigned long> processedFaces{};
+        auto insertIfAbsent = [&triangles, &processedFaces](const auto &planeEvent) {
+            const auto faceIndex{planeEvent.faceIndex};
+            if (processedFaces.find(faceIndex) == processedFaces.end()) {
+                processedFaces.insert(faceIndex);
+                triangles.push_back(faceIndex);
+            }
+        };
+        std::for_each(eventList.cbegin(), eventList.cend(), insertIfAbsent);
+        triangles.shrink_to_fit();
```

Comment:
Nah, low-effort solution. Just keep it in mind that this might be an interesting metric for data-structure-memory things.

If you profile, then I would just recommend a dirty approach and compare "Before/ After" by uncommenting everything related to these operations via "STR + R" (i.e. the replace operation) 😉

Reply:
The flags are a bit overengineered in my opinion because this is a production code (and less a benchmarking one). So, we want to keep the code base concise.

But what I strongly recommend (I did a similar approach for my bachelor thesis).

Put everything **experimental** in a separate branch for documentation/ replication reasons, e.g. `experimental-no-memory-optimization`.
Then, we keep `main` clear, but you can still quickly switch (and don't have a potential ambiguous setup).